crates/rooch-relayer/src/actor/bitcoin_relayer.rs

```diff
@@ -244,3 +260,52 @@ impl Handler<GetReadyL1TxsMessage> for BitcoinRelayer {
         self.get_ready_l1_txs()
     }
 }
+
+pub struct BitcoinReorgAwareStore {
+    block_store_dir: PathBuf,
+    recent_blocks_map: IndexMap<u64, BlockHash>,
+    aware_height: usize,
+    rpc_client: BitcoinClientProxy,
+}
+
+impl BitcoinReorgAwareStore {
+    pub fn new(
+        block_store_dir: PathBuf,
+        aware_height: usize,
+        rpc_client: BitcoinClientProxy,
+    ) -> Self {
+        Self {
+            block_store_dir,
+            recent_blocks_map: IndexMap::with_capacity(aware_height),
+            aware_height,
+            rpc_client,
+        }
+    }
+
+    pub async fn insert_or_replace(
```

Comment:
Should we put the BitcoinReorgAwareStore logic into BitcoinClientActor? Make the read-and-write operation in the same mod.

Reply:
In present, client is read-only too (for connect bitcoin node). And it's hard for a client to know which block has been consumed and could regard as reorg for Rooch

Reply:
If the BitcoinClientActor saves all blocks as cache data, maybe we can remove this function.